app/javascript/mastodon/components/status_content.jsx

```diff
@@ -126,6 +126,7 @@ class StatusContent extends PureComponent {
         link.setAttribute('href', `/tags/${link.text.replace(/^#/, '')}`);
       } else {
```

Comment:
as noted in https://github.com/mastodon/mastodon/pull/26917#issuecomment-1721141573 this doesn't play nice with lemmy posts. Something like the following should improve that.

```suggestion
     } else if (status.get('uri') === link.href) {
        // the link points to the source of the status. this happens e.g. with lemmy posts.
        // there is no use in trying to open the url on this instance as it's already opened.
        link.setAttribute('title', link.href);
        link.classList.add('unhandled-link');
      } else {
```

(like i did in https://github.com/mastodon/mastodon/pull/28762/commits/7c957978ac2bb8d0ff2a259dc0c5527aac66c717 )